core/src/types/eth_bridge_pool.rs

```diff
@@ -84,51 +178,82 @@ impl PendingTransfer {
+            kind: *appendix.kind,
+            asset: event.asset,
+            recipient: event.receiver,
+            sender: (*appendix.sender).clone(),
+            amount: event.amount,
+        };
+        let gas_fee = (*appendix.gas_fee).clone();
+        Self { transfer, gas_fee }
+    }
 }
 
-impl From<PendingTransfer> for ethbridge_structs::Erc20Transfer {
-    fn from(pending: PendingTransfer) -> Self {
+impl From<&PendingTransfer> for ethbridge_structs::Erc20Transfer {
+    fn from(pending: &PendingTransfer) -> Self {
+        let HashDigest(namada_data_digest) = pending.appendix().checksum();
         Self {
-            kind: pending.transfer.kind as u8,
             from: pending.transfer.asset.0.into(),
             to: pending.transfer.recipient.0.into(),
             amount: pending.transfer.amount.into(),
-            fee_from: pending.gas_fee.payer.to_string(),
-            fee: pending.gas_fee.amount.into(),
-            sender: pending.transfer.sender.to_string(),
+            namada_data_digest,
         }
     }
 }
 
-impl Encode<8> for PendingTransfer {
-    fn tokenize(&self) -> [Token; 8] {
+impl From<&PendingTransfer> for TransferToEthereumEvent {
+    fn from(pending: &PendingTransfer) -> Self {
+        Self {
+            amount: pending.transfer.amount,
+            asset: pending.transfer.asset,
+            receiver: pending.transfer.recipient,
+            checksum: pending.appendix().checksum(),
+        }
+    }
+}
+
+impl Encode<6> for PendingTransfer {
+    fn tokenize(&self) -> [Token; 6] {
         // TODO: This version should be looked up from storage
-        let version = Token::Uint(1.into());
+        let version = Token::Uint(VERSION.into());
         let namespace = Token::String(NAMESPACE.into());
         let from = Token::Address(self.transfer.asset.0.into());
-        let fee = Token::Uint(self.gas_fee.amount.into());
         let to = Token::Address(self.transfer.recipient.0.into());
         let amount = Token::Uint(self.transfer.amount.into());
-        let fee_from = Token::String(self.gas_fee.payer.to_string());
-        let sender = Token::String(self.transfer.sender.to_string());
-        [version, namespace, from, to, amount, fee_from, fee, sender]
+        let checksum = Token::FixedBytes(self.appendix().checksum().0.into());
+        [version, namespace, from, to, amount, checksum]
     }
 }
 
-impl From<&TransferToEthereumEvent> for PendingTransfer {
-    fn from(event: &TransferToEthereumEvent) -> Self {
-        let transfer = TransferToEthereum {
-            kind: event.kind,
-            asset: event.asset,
-            recipient: event.receiver,
-            sender: event.sender.clone(),
-            amount: event.amount,
-        };
-        let gas_fee = GasFee {
-            amount: event.gas_amount,
-            payer: event.gas_payer.clone(),
-        };
-        Self { transfer, gas_fee }
+impl Encode<6> for TransferToEthereumEvent {
```

Comment:
Why do we still use this trait? I thought the ethereum_structs crate allowed us to remove this.

Reply:
`TransferToEthereum` is a custom type defined by us, long before `ethbridge-rs` was even a thing. it's basically legacy code... but there are some conversions between `ethabi` native types and Namada native types, when we translate the types generated by `ethbridge-rs` to an `EthereumEvent` enum.